### ISSUE TITLE
Other things to look into

### ISSUE THREAD

**Issue by acheong08** [acheong08/apple-corelocation-experiments#1]
**https://gsp85-ssl.ls.apple.com/wifi_request**
 - Comes from `com.apple.geod`
 - Sends MAC address and gets a `fixed64` back. (Does 1804284399684943872 mean anything to anyone?)

**Comment by acheong08** [comment]
```
$ strings -a dyld_shared_cache_x86_64h | rg -i "gsp85-carry.ls.apple.com"

https://gsp85-carry.ls.apple.com/wifi_request_tile
```

Can't find `wifi_request` in any binaries but found something close in `dyld_shared_cache_x86_64h`. Need to figure out which of the many dylds contained is the one that gave that

**Comment by acheong08** [comment]
Some magic numbers:
167797871, 3982925012049145399, 6124895493240913920

They seem to be shared across requests for different BSSIDs. Might encode my location but it's not right (it's somehow in the middle of the sea). I'm guessing the first number is an offset somehow

**Comment by acheong08** [comment]
The first URL seems to send the BSSID of whatever network you're connected to along with metadata. Can reliably reproduce by joining new network.

**Comment by acheong08** [comment]
This is too fucking hard. They generate C code from their protobuf definitions without embedding the proto file itself and stripped all symbols from GeoServices

**Comment by acheong08** [comment]
![image](https://github.com/acheong08/apple-corelocation-experiments/assets/36258159/667841e2-5689-4b8b-bd78-c902be62429f)

Not even Apple knows what the fields mean

**Comment by acheong08** [comment]
Update on `wifi_request` endpoint. `X-tilekey` seems to be morton code encoded with a different index. More experimentation required but I need sleep

**Comment by acheong08** [comment]
It is indeed morton encoded.

(y-, x+)
Basically the further north, the more y decreases and the further east, the more x increases

**Comment by AK47-2020** [comment]
Hi, great tool. Is there a way to get the vendor name matching a MAC address, possibly as a display option?

**Comment by acheong08** [comment]
Yup. Just did. Note that the CLI is just for me to play around and experiment with things more easily. Don't expect it to be coherent.

**Comment by AK47-2020** [comment]
Thanks.
How do I extract a list of MAC addresses from, say, a square area?
Also, it would be good to have a dialog box to change defaut coordinates. Indeed, you can them manually.

Envoyé avec la messagerie sécurisée [Proton Mail.](https://proton.me/)

**Comment by h2oFanti** [comment]
> It is indeed morton encoded.
> 
> (y-, x+) Basically the further north, the more y decreases and the further east, the more x increases

Is this still the case or was it resolved in related REAME.md "Update 2"? I stumbled across something that reminded me of this project: [WGS 84](https://en.wikipedia.org/wiki/World_Geodetic_System#WGS_84)

Pardon this intrusion if its unrelated or you've checked that, I hardly understand coordinate systems and do understand various ways this type of relation can manifest, but appreciate this project and the work being done.

**Comment by acheong08** [comment]
> Is this still the case or was it resolved in related REAME.md "Update 2"?

Yup. I found the solution and it works fine now.
